packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart

```diff
@@ -655,19 +652,19 @@ Information about project "Runner":
     expect(info.buildConfigurationFor(BuildInfo.release, 'Runner'), 'Release');
   });
 
-  testWithoutContext('scheme for project with custom schemes is matched against flavor', () {
+  testUsingContext('scheme for project with custom schemes is matched against flavor', () {
     final XcodeProjectInfo info = XcodeProjectInfo(
       <String>['Runner'],
       <String>['Debug (Free)', 'Debug (Paid)', 'Release (Free)', 'Release (Paid)'],
       <String>['Free', 'Paid'],
       logger,
     );
 
-    expect(info.schemeFor(const BuildInfo(BuildMode.debug, 'free', treeShakeIcons: false)), 'Free');
-    expect(info.schemeFor(const BuildInfo(BuildMode.profile, 'Free', treeShakeIcons: false)), 'Free');
-    expect(info.schemeFor(const BuildInfo(BuildMode.release, 'paid', treeShakeIcons: false)), 'Paid');
-    expect(info.schemeFor(BuildInfo.debug), isNull);
-    expect(info.schemeFor(const BuildInfo(BuildMode.debug, 'unknown', treeShakeIcons: false)), isNull);
+    expect(info.schemeFor(flavor: 'Free'), 'Free');
```

Comment:
I don't see a test that's passing in a flavor and hostAppProjectName.  Imagine you had a bug that preferred the app name over the flavor.